### PR TITLE
Update changelog [ECR-3516]

### PR DESCRIPTION
**Pull request by popzxc**
Hooray, I guess?

Once this gets merged, feel free to turn on your "please-update-a-changelog"-o-meter.

[A link to the rendered version of `CHANGELOG.md`](https://github.com/popzxc/exonum/blob/update-changelog/CHANGELOG.md)

This update schema:
- This is not a complete migration guide / documentation. Only a short description was added with links to the docs/examples.
- Missing changelog entries for non-dymanic-services changes were added.
- For dynamic services, a four-level description is given:
  1. A short overview.
  2. A short migration guide (not really a guide, more of "where you can find the info" section).
  3. A list of important PRs within implementing the feature.
  4. A list of ALL PRs related to this feature. "Garbage" PRs (e.g. "merge master") are filtered out. However, PRs from `3` are included.
- __Important:__ Inserted links are adapted for the release. They refer to the 0.13 release artifacts, so they don't work now, but will become correct on the release.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -59,9 +75,351 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
...
+
+- Access to the database is now isolated for services.
+  Service can't get a write access to other service or blockchain schema.
+
+- Transactions don't have an `execute` method anymore. Instead, service defines
+  and implements an interface trait which contain all the business logic.
+
+- Services aren't launched at the node start by default. For launching a
+  service, consider using an [`exonum-launcher`][launcher] tool.
+
```
Comment (alekseysidorov) [inline CHANGELOG.md]: Nit: I think that we should add paragrapth about services with configuration.

File: CHANGELOG.md [hunk 2]
```diff
@@ -59,9 +75,351 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
...
+  [launcher]: https://github.com/exonum/exonum-launcher
+
+#### Important PRs for Dynamic Services
+
+<!-- markdownlint-disable no-inline-html -->
+<details>
+    <summary>Under the spoiler you can find list of pull requests
```
Comment (Maria-Nosyk) [inline CHANGELOG.md]:
```suggestion
    <summary>Below you can find a list of pull requests
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -59,9 +75,351 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
...
+    which have significant meaning for the Dynamic Services
```
Comment (Maria-Nosyk) [inline CHANGELOG.md]:
```suggestion
    which have significant meaning for the implementation of the Dynamic Services
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -59,9 +75,351 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
...
+    feature implementation.
```
Comment (Maria-Nosyk) [inline CHANGELOG.md]:
```suggestion
    feature.
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -59,9 +75,351 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
...
+- #1492: Do start and initialize service at single step [ECR-3222]
+
+- #1537: Finalize Exonum-derive macros [ECR-3800]
+
+- #1538: Supervisor modes [ECR-3794] [ECR-3771]
+
+</details>
+<!-- markdownlint-enable no-inline-html -->
+
+#### Dynamic services implementation full history
```
Comment (Maria-Nosyk) [inline CHANGELOG.md]:
```suggestion
#### Full History of the Dynamic Services Implementation 
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -59,9 +75,351 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
...
+
+<!-- markdownlint-disable no-inline-html -->
+<details>
+    <summary>Under the spoiler you can find list of all pull requests related
```
Comment (Maria-Nosyk) [inline CHANGELOG.md]:
```suggestion
    <summary>Below you can find a list of all pull requests related
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -59,9 +75,351 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
...
+    to the Dynamic Services feature implementation.
```
Comment (Maria-Nosyk) [inline CHANGELOG.md]:
```suggestion
    to the implementation of the Dynamic Services feature.
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -93,18 +453,36 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
   - For `Hash` keys both map variants works the same, because `ObjectHash`
   implementation for `Hash` returns the hash itself.
 
+#### exonum-cli
+
+- `exonum-cli` now provides a public reexport of `structopt` crate (#1461).
+
 ### Internal improvements
```
Comment (Maria-Nosyk) [inline CHANGELOG.md]:
```suggestion
### Internal Improvements
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -93,18 +453,36 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
...
 
 #### exonum
 
 - `system/v1/shutdown` endpoint has been modified and now accepts empty POST
   requests. (#1526)
 
+- `keys` module was moved into `exonum-keys` crate. (#1497)
```
Comment (Maria-Nosyk) [inline CHANGELOG.md]:
```suggestion
- `keys` module has been moved into `exonum-keys` crate. (#1497)
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -93,18 +453,36 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
...
+
 #### exonum-merkledb
 
 - `ProofListIndex::extend()` method has been refactored, leading to up to 10x
   performance improvements for large lists. (#1455)
 
+- Proofs building mechanisms were heavily refactored. (#1460)
```
Comment (Maria-Nosyk) [inline CHANGELOG.md]:
```suggestion
- Proofs building mechanisms have been heavily refactored. (#1460)
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -93,18 +453,36 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
...
+
+### Bug Fixes
+
+#### exonum
+
+- Localhost ports 8080/8081 became allowed in CORS within `run-dev` mode. (#1415)
```
Comment (Maria-Nosyk) [inline CHANGELOG.md]:
```suggestion
- Localhost ports 8080/8081 are now allowed in CORS within the `run-dev` mode. (#1415)
```

**Pull request merged**